backend/app/core/exceptions.py

```diff
@@ -16,3 +15,15 @@
     status_code=401,
     detail="You do not have permission for this action."
 )
+
+
+class SecretsModuleNotFoundError(ImportError):
```

Comment:
Not a big fan of sub-classing standard errors. I would suggest to have:

* A common class for all Swiple exception (e.g. `SwipleError`)
* A common class for secrets-related errors, e.g. `SecretsError`, inheriting from `SwipleError`
* The three errors we have here, inheriting from `SecretsError`.

The good thing with this is that we could implement catchers at several levers (all Swiple errors, all secrets errors...)

Reply:
Do you like the way that Superset handles their exceptions?

https://github.com/apache/superset/blob/cbadb8773f06ca62ad4d9ebf04df211af3de9853/superset/exceptions.py

Reply:
It's nicely done indeed; but maybe a bit too complex for our needs right now.

Reply:
Works nicely. Are these updates what you expected?

Reply:
Yeah, looks great!